$OpenBSD: patch-crypt-gpgme_c,v 1.2 2015/09/09 18:40:55 sthen Exp $

gpgme's gpgme_subkey_t->timestamp is an unsigned long and can't be passed
directly to localtime which takes a time_t.

http://dev.mutt.org/trac/changeset/6498/crypt-gpgme.c

--- crypt-gpgme.c.orig	Wed Sep  9 20:33:56 2015
+++ crypt-gpgme.c	Wed Sep  9 20:34:28 2015
@@ -2018,6 +2018,7 @@ static int pgp_gpgme_extract_keys (gpgme_data_t keydat
   char date[STRING];
   int more;
   int rc = -1;
+  time_t tt;
 
   if ((err = gpgme_new (&tmpctx)) != GPG_ERR_NO_ERROR)
   {
@@ -2081,7 +2082,8 @@ static int pgp_gpgme_extract_keys (gpgme_data_t keydat
       len = mutt_strlen (subkey->keyid);
       if (len > 8)
         shortid += len - 8;
-      strftime (date, sizeof (date), "%Y-%m-%d", localtime (&subkey->timestamp));
+      tt = subkey->timestamp;
+      strftime (date, sizeof (date), "%Y-%m-%d", localtime (&tt));
 
       if (!more)
         fprintf (*fp, "%s %5.5s %d/%8s %s %s\n", more ? "sub" : "pub",
